BatteryStateManagement_proj.c

```diff
@@ -0,0 +1,83 @@
+/*****************************************************************************/
+/***    \file        BatteryStateManagement_proj.c
+***     \author      Naveen Balachandar (RBEI/ECK3) 
+***
+***     \brief       File to monitor the battery condition based required properties
+/*****************************************************************************/
+/*------ standard includes -------*/
+#include <stdio.h>
+#include <assert.h>
+
+/*---------------------------------------------------------------------------*/
+/*     FUNCTION:    batteryCondMonitor_i
+ */
+/*!    \brief       predecting  the battery states
+ * 
+ *     \param       current temperature/soc/chargeRate ,respective min and max rate ,index 
+ *     \returns     batterystate -low/high/normal
+ *
+*//*------------------------------------------------------------------------*/
+int batteryCondMonitor_i(float batteryParameter ,float minRange, float maxRange,int batParIndex)
+{
+  /*Battery par printed for ref*/
+  char batPar[3][10] = {"temp","soc","chargerate"};
+  
+  if(batteryParameter < minRange)   /*Min range valid*/
+  {
+   printf("Battery parameter %s is low!\n",batPar[batParIndex]);
+   return 0;
+  }
+  else if (batteryParameter > maxRange) /*Max range valid*/
+  {
+   printf("Battery parameter %s is High !\n",batPar[batParIndex]);
+   return 0;
+  }
+  else
+  {
+   printf("Battery parameter %s is Normal !\n",batPar[batParIndex]);
+   return 1;
+  }
+}
+
+/*---------------------------------------------------------------------------*/
+/*     FUNCTION:    batteryStateValidation_i
+ */
+/*!    \brief       Monitoring the battery states
+ * 
+ *     \param       temperature,soc,chargeRate
+ *     \returns     batterystate -OK/NOK
+ *
+*//*------------------------------------------------------------------------*/
+int batteryStateValidation_i(float temperature, float soc, float chargeRate)
+{ 
+ 
+ int retTempStat_i, retsocStat_i,retchargeStat_i;
+  
+  /*Validation done with cureent battery parameter,Min and max range*/
+  retTempStat_i   = batteryCondMonitor_i(temperature,0,45,0);
+  retsocStat_i    = batteryCondMonitor_i(soc,20,80,1);
+  retchargeStat_i = batteryCondMonitor_i(chargeRate,0,0.8,2);
+  
+  /*return battery state ok /nok*/
+  return ((retTempStat_i & retsocStat_i) & retchargeStat_i);
+}
+
+void main() 
```

Comment:
Nice tests :)
But looks like the 'lesser' part is never tested? How would you recognize if any other conditions are missed?